x-pack/plugins/security/server/authentication/authentication_service.ts

```diff
@@ -106,8 +106,20 @@ export class AuthenticationService {
     });
 
     http.registerAuth(async (request, response, t) => {
-      // If security is disabled continue with no user credentials and delete the client cookie as well.
+      if (!license.isLicenseAvailable()) {
+        this.logger.error('License is not available, authentication is not possible.');
+        return response.customError({
+          body: 'License is not available.',
+          statusCode: 503,
+          headers: { 'Retry-After': '30' },
+        });
+      }
+
+      // If security is disabled, then continue with no user credentials.
       if (!license.isEnabled()) {
```

Comment:
__note:__ technically we don't need this `if` in 7.11+ anymore, but I'd prefer to keep it for now.